src/launchpad/service.py

```diff
@@ -78,58 +127,396 @@ def handle_kafka_message(self, payload: PreprodArtifactEvents) -> None:
+        self, sentry_client: SentryClient, artifact_id: str, project_id: str, organization_id: str, error_msg: str
+    ) -> None:
+        """Handle download error by logging and updating artifact."""
+        logger.error(error_msg)
+        # Extract just the error details for the database, the enum prefix will be added automatically
+        detailed_error = error_msg.split(": ", 1)[1] if ": " in error_msg else error_msg
+        self._update_artifact_error(
+            sentry_client,
+            artifact_id,
+            project_id,
+            organization_id,
+            ERROR_CODE_ARTIFACT_PROCESSING_ERROR,
+            ProcessingErrorMessage.DOWNLOAD_FAILED,
+            detailed_error,
+        )
+        raise RuntimeError(error_msg)
+
+    def _download_artifact(
+        self,
+        sentry_client: SentryClient,
+        artifact_id: str,
+        project_id: str,
+        organization_id: str,
+    ) -> tuple[bytes, int]:
+        """Download artifact from Sentry and validate response."""
+        logger.info(f"Downloading artifact {artifact_id}...")
 
         try:
-            # Wait for shutdown signal
-            await self._shutdown_event.wait()
+            download_result = sentry_client.download_artifact(
+                org=organization_id, project=project_id, artifact_id=artifact_id
+            )
+
+            if "error" in download_result:
+                error_category, error_description = categorize_http_error(download_result)
+                self._handle_download_error(
+                    sentry_client,
+                    artifact_id,
+                    project_id,
+                    organization_id,
+                    f"Failed to download artifact ({error_category}): {error_description}",
+                )
+
+            if not download_result.get("success"):
+                self._handle_download_error(
+                    sentry_client,
+                    artifact_id,
+                    project_id,
+                    organization_id,
+                    f"Download was not successful: {download_result}",
+                )
+
+            file_content = download_result["file_content"]
+            file_size = download_result["file_size_bytes"]
+
+            logger.info(f"Downloaded artifact {artifact_id}: {file_size} bytes ({file_size / 1024 / 1024:.2f} MB)")
+            return file_content, file_size
+
+        except Exception as e:
+            if not isinstance(e, RuntimeError):
+                detailed_error = str(e)
+                self._update_artifact_error(
+                    sentry_client,
+                    artifact_id,
+                    project_id,
+                    organization_id,
+                    ERROR_CODE_ARTIFACT_PROCESSING_ERROR,
+                    ProcessingErrorMessage.DOWNLOAD_FAILED,
+                    detailed_error,
+                )
+            raise
+
+    def _save_to_temp_file(self, file_content: bytes, artifact_id: str) -> str:
```

Comment:
The handling of tempfile is a little odd. Ideally we would use the built in cleanup behaviour of tempfile rather than manually cleaning up ourselves. See comment at use.